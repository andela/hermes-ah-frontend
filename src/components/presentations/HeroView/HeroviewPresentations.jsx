import React from 'react';
import { Link } from 'react-router-dom';
import { Button } from 'semantic-ui-react';
import './hero-view.scss';

const HeroView = () => {
  return (
    <div className="heroAll">
      <div className="overlay">
        <div className="siteTexts">
          <h1 className="siteName">Authors Haven</h1>
          <p className="siteTitle">Be part of a solution</p>
          <p className="siteTitle" id="siteTitleSmall">
            A home for bright minds to make their contributions to the world and
            profer solutions
          </p>
          <div>
            <Link to="login">
              <Button primary>Explore</Button>
            </Link>
          </div>
        </div>
      </div>
    </div>
  );
};

export default HeroView;
