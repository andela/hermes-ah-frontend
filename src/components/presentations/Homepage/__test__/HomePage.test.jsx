import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import HomePage from '../Homepage';
import HeroView from '../../HeroView/HeroviewPresentations';

const middleware = [thunk];
const mockStore = configureMockStore(middleware);
const store = mockStore({});

const HomePageComponent = (
  <Provider store={store}>
    <HomePage
      getAllArticles={() => 'articles'}
      articles={{ articleData: [{}] }}
      isLoadingReducer={{ loader: false }}
    />
  </Provider>
);

describe('HomePage component', () => {
  it('should match snapshot', () => {
    const wrap = HomePageComponent;
    expect(wrap).toMatchSnapshot();
  });

  it('should render without crashing', () => {
    const wrapper = shallow(HomePageComponent);
    expect(wrapper.find(<HeroView />));
  });
});
