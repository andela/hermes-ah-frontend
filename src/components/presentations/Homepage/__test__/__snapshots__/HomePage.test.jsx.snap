// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component should match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Homepage
    articles={
      Object {
        "articleData": Array [
          Object {},
        ],
      }
    }
    getAllArticles={[Function]}
    isLoadingReducer={
      Object {
        "loader": false,
      }
    }
    user={null}
  />
</Provider>
`;
